pymc3/distributions/discrete.py

```diff
@@ -438,17 +441,17 @@ class DiscreteUniform(Discrete):
 
     def __init__(self, lower, upper, *args, **kwargs):
         super(DiscreteUniform, self).__init__(*args, **kwargs)
-        self.lower = tt.floor(lower).astype('int32')
-        self.upper = tt.floor(upper).astype('int32')
+        self.lower = intX(np.floor(lower))
```

Comment:
Having theano tensor as distribution bound inputs? Is that intended? I should have known that in advance! If yes we can forget about this whole PR ...